plugins/inputs/suricata/suricata.go

```diff
@@ -150,27 +154,35 @@ func flexFlatten(outmap map[string]interface{}, field string, v interface{}, del
 		}
 	case float64:
 		outmap[field] = v.(float64)
+	case string:
+		outmap[field] = v.(string)
 	default:
 		return fmt.Errorf("unsupported type %T encountered", t)
 	}
 	return nil
 }
 
-func (s *Suricata) parse(acc telegraf.Accumulator, sjson []byte) {
-	// initial parsing
-	var result map[string]interface{}
-	err := json.Unmarshal(sjson, &result)
-	if err != nil {
-		acc.AddError(err)
+func (s *Suricata) parseAlert(acc telegraf.Accumulator, result map[string]interface{}) {
+	if _, ok := result["alert"].(map[string]interface{}); !ok {
+		s.Log.Debug("The 'alert' sub-object does not have required structure")
```

Comment:
It's only cosmetics but you should change to
```suggestion
		s.Log.Debug("'alert' sub-object does not have required structure")
```